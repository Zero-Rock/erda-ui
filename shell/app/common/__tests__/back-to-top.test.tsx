import React from 'react';
import BackToTop from '../components/back-to-top';
import { describe, it, jest } from '@jest/globals';
import { act } from 'react-dom/test-utils';
import { render } from 'react-dom';


describe('BackToTop', () => {
  Element.prototype.scrollTo = function (opt? :ScrollToOptions|number) {
    if (typeof opt !== 'number') {
      const { top, left } = opt as ScrollToOptions;
      this.scrollTop = top || 0;
      this.scrollLeft = left || 0;
    }
  };
  it('should back to top', async () => {
    document.body.innerHTML = '<div id="main" style="height: 400px; overflow-y: auto"></div>';
    const div = document.getElementById('main') as HTMLDivElement;
    const scrollToSpy = jest.spyOn(window, 'scrollTo').mockImplementation((x: number, y: number) => {
      div.scrollLeft = x;
      div.scrollTop = y;
    });
    act(() => {
      render(
        <div id="child" style={{ height: '1000px' }}>
          <BackToTop />
        </div>, div
      );
    });
    window.scrollTo(0, 500);
    expect(div.scrollTop).toBe(500);
    act(() => {
      div.dispatchEvent(new Event('scroll'));
    });
    const button = document.querySelector('.scroll-top-btn') as Element;
    act(() => {
      button.dispatchEvent(new MouseEvent('click', { bubbles: true }));
    });
    expect(div.scrollTop).toBe(0);
    scrollToSpy.mockRestore();
  });
});
