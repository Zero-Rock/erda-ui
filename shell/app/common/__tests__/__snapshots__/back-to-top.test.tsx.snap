// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackToTop should  1`] = `<BackToTop />`;

exports[`BackToTop should act 1`] = `null`;
